### ISSUE TITLE
Filter only works on single collection at a time

### ISSUE THREAD

**Issue by emko** [darktable-org/darktable#13116]
I want to see all my images from two different hard drives i added them like this

![image](https://user-images.githubusercontent.com/1070310/206879574-6dfe9af4-ebe4-4e45-88ea-44b1a7224494.png)

now when i set say 5 star rating only, it seems to only show 5 stars from one of these folders and the rest of the images from the other. 

I would expect it to filter all the images to what the filters are set to

to reproduce 
add a second folder via Add more images
use the collection filter to filter the search (5 star)
the filter is being applied only to the second folder

**Comment by ptilopteri** [comment]
read about collection filters, the left panel window directly below "collections"

**Comment by emko** [comment]
> read about collection filters, the left panel window directly below "collections"

i am not following? what do you mean read about? i have the filter set to 5stars 

![image](https://user-images.githubusercontent.com/1070310/206880246-7fd359e8-ab0e-4e12-866f-812eff1a8962.png)

i though it was different folders its just random... or i am not understanding how dt filtering/collections work, all those images are in same folder :(

**Comment by emko** [comment]
found a way to have it working how i want it, i did a single folder collection with /mnt/photography* or /mnt/HDBackup* in the input seems to work fine just the collection input becomes blank

**Comment by ptilopteri** [comment]
then you should close this issue

**Comment by emko** [comment]
> then you should close this issue

you sure? your saying user should know SQL to figure out something basic as making filtering work correctly ?

**Comment by guille2306** [comment]
> then you should close this issue

The issue is real, I can reproduce on DT 4.0.1 under Linux.

- If you have conditions "a OR b" in collections, and you have "c" condition in filters, that is taken as "a OR (b AND c)". It should be "(a OR b) AND c"
- If you have conditions "a OR b OR c" in collections, and you have "d AND e" condition in filters, that is taken as "a OR b OR (c AND d AND e)". It should be "(a OR b OR c) AND (d AND e)"
- etc.

**Comment by guille2306** [comment]
Inversely

- if you have condition "a" in collections, and conditions "b OR c" in filters, that is taken as "(a AND b) OR c"

This is not only with folders/filmrolls and stars, it applies to any condition in both modules.

I would say that boolean operation application when conditions in both collections and filter modules are used is broken (and would advise to change the title of the issue to match this).

**Comment by MStraeten** [comment]
there's no terminating bracket after collection filters at the proper place
```
SELECT DISTINCT mi.id FROM (
SELECT  id, ...  
FROM main.images 
AS mi  ) AS mi 
WHERE  
 (flags & 256) != 256  
 AND  (
	1=1	AND (film_id IN ( SELECT id FROM main.film_rolls WHERE folder LIKE '/folder1')) 
		OR (film_id IN ( SELECT id FROM main.film_rolls WHERE folder LIKE '/folder 2')) 
```
not here 	
```
AND (flags & 8 == 0 AND flags & 7 >= 2)
```
but here giving a different result.

Unfortunately not that easy since the collection module itself can contain rating or further filters without being able to bracket them properly.
So it lacks a consistent UX design of the whole collection&filter shebang. 

minimum solution: bracketing the boolean expression of collection module and filter&sort module according to a simple rule: first define a collection, then filter/sort the whole result.

**Comment by guille2306** [comment]
> minimum solution: bracketing the boolean expression of collection module and filter&sort module according to a simple rule: first define a collection, then filter/sort the whole result.

This would also be my expectation: first _collect_ images, then _filter_ them. So: "(collection conditions) AND (filter conditions)".

The fact that collections and filters overlap in functionality should be a separate problem, and not solved by trying to mix the sql calls.

**Comment by AlicVB** [comment]
> > minimum solution: bracketing the boolean expression of collection module and filter&sort module according to a simple rule: first define a collection, then filter/sort the whole result.
> 
> This would also be my expectation: first _collect_ images, then _filter_ them. So: "(collection conditions) AND (filter conditions)".
> 
> The fact that collections and filters overlap in functionality should be a separate problem, and not solved by trying to mix the sql calls.

Completely agree. I'll fix that. (but that won't be for 4.2 due to the freeze I fear)

**Comment by Nilvus** [comment]
Collection filters were a first step because it's a huge work and there's still improvement to do. A consistent UX design had been discussed for 2023 with @AlicVB. That's why you could have seen on pixls.us threads from @AlicVB few weeks ago. One of big steps we hope for 2023 is to have a better collection & filter design. It's not a simple thing though. All mockups for such big rewrite are welcomed.

If you have ideas, please post them on same post here: #13065 (I already ask as you can see for such there). I suggest before, as far as possible, to read following threads:
- https://discuss.pixls.us/t/discussion-around-collection-filters-improvements/32489
- https://discuss.pixls.us/t/discussion-around-new-filters-widgets/32490

**Comment by Nilvus** [comment]
@AlicVB: as I post quite same time as you. I think this should go with the collection & filters rewrite if possible.

**Comment by TurboGit** [comment]
> Completely agree. I'll fix that. (but that won't be for 4.2 due to the freeze I fear)

A fix for 4.2 is still welcome as the freeze is not yet done.
